fleetctl/destroy.go

```diff
@@ -33,9 +33,14 @@ Destroyed units are impossible to start unless re-submitted.`,
 }
 
 func runDestroyUnits(args []string) (exit int) {
-	for _, v := range args {
-		name := unitNameMangle(v)
-		err := cAPI.DestroyUnit(name)
+	units, err := findUnits(args)
+	if err != nil {
+		stderr("%v", err)
+		return 1
+	}
+
+	for _, v := range units {
+		err := cAPI.DestroyUnit(v.Name)
```

Comment:
@jonboulle oh for the race condition you have a point here, yes. Adding checks for Destroy will fix it of course, it could be done without removing findUnits() call since my aim was to have more consistency and avoid hard coding logic for each command. @antrik will follow up with a fix.